app/models/user.rb

```diff
@@ -234,6 +234,7 @@ def name(use_email = true)
   #
   # Returns UserIdentifier
   def identifier_for(scheme)
+    scheme = scheme.instance_of?(IdentifierScheme) ? scheme.name : scheme
```

Comment:
it would likely be better to do this in the Identifier.by_scheme_name method itself.